client/src/components/general/VirtualizedList.tsx

```diff
@@ -0,0 +1,80 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+import styles from './VirtualizedList.module.scss';
+import { TransitionGroup, CSSTransition } from 'react-transition-group';
+import { ListItem } from "../../models/ListItem";
+
+type Props = {
+    items: ListItem[],
+    itemHeight: number
+}
+
+export function VirtualizedList(props: Props) {  
+  const { items, itemHeight } = props;
+  const [selectedItem, setSelectedItem] = useState<number | null>(null);
+  const [visibleItems, setVisibleItems] = useState<ListItem[]>([]);
+  const [scrollTop, setScrollTop] = useState(0);
+  
+  const container = useRef<HTMLDivElement>(null);
+  
+  const changeBackgroundColor = useCallback((id: number) => {
```

Comment:
I would name this function `selectItem` since this is what it's doing in practice. it makes the use of it more indicative.

Reply:
you're right, changed it :)